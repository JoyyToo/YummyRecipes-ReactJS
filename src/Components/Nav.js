import React, {Component} from 'react';
import { Link } from 'react-router-dom';
import {notify} from 'react-notify-toast'
import axios from 'axios';
import * as constant from "./constant";
import './Landing.css'

class Nav extends Component {

    handleLogout = (event) => {
        const token = window.localStorage.getItem('token')

        axios({
            url: `${constant.URL}/auth/logout`,
            method: 'post',
            headers: {
                Authorization: `Bearer ${token}`,
                'Content-Type': 'multipart/form-data',
            }
        })

        .then((response) => {   
            window.localStorage.clear()      
            console.log(response.data.message)
            notify.show(response.data.message, 'success', 4000)
            this.props
            .history
            .push('/');
        })
        .catch((error) => {
            console.log(error.response.data.message)
            notify.show(error.response.data.message, 'error', 4000);
        });
    }

    render() {
        const Accesstoken = window.localStorage.token

        return (
            
            <div className="container">
            <Link to="/" className="navbar-brand" style={{ textDecoration: 'none' }}><h2>YUMMY

            { !Accesstoken ? (
                        ''
                    ) : (

                        <Link to="/categories" className="navbar-brand" style={{ float: 'right', marginRight: 40, textDecoration: 'none' }}
                         >CATEGORIES
                        <Link to="/" className="navbar-brand" style={{ float: 'right', marginRight: 40,  marginLeft: 30,  textDecoration: 'none' }}
                              onClick={(event => this.handleLogout(event))} >LOGOUT</Link></Link>

                    )}

            <a className="navbar-brand" style={{ float: 'right',  marginRight: 20, color: 'white'}}>           
                Hello
                { !Accesstoken ? (
                    ' Guest'
                ) : (
                    ' ' + localStorage.getItem('username')
                )}
            </a>
            </h2>
            </Link>

            </div>
        )
    }}

    export default Nav;
