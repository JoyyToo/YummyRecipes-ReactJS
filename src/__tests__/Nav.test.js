import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Nav from '../Components/Nav';

describe('Nav component', () => {
  const wrapper = shallow(<Nav />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders 1 <Nav/> component', () => {
    const component = shallow(<Nav />);
    expect(component).toHaveLength(1);
  });

  it('it returns props correctly', () => {
    const component = shallow(<Nav name="app" />);
    expect(component.instance().props.name).toBe('app');
  });

  it('contains div', () => {
    expect(wrapper.find('div')).toHaveLength(1);
  });

  it('contains h2', () => {
    expect(wrapper.find('h2')).toHaveLength(1);
  });

  it('contains Link', () => {
    expect(wrapper.find('Link')).toHaveLength(1);
  });
});
