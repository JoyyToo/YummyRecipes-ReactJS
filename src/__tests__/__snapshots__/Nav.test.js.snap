// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders properly 1`] = `
<div
  className="container"
>
  <Link
    className="navbar-brand"
    replace={false}
    to="/"
  >
    <h2>
      YUMMY
    </h2>
  </Link>
</div>
`;
